query-engine/connectors/sql-query-connector/src/filter_conversion.rs

```diff
@@ -546,16 +685,16 @@ fn lower_if(expr: Expression<'static>, cond: bool) -> Expression<'static> {
     }
 }
 
-fn convert_first_value<'a>(fields: &[ScalarFieldRef], value: PrismaValue) -> Value<'a> {
-    fields.first().unwrap().value(value)
-}
-
 fn convert_value<'a>(field: &ScalarFieldRef, value: PrismaValue) -> Value<'a> {
     field.value(value)
 }
 
+fn convert_first_value<'a>(fields: &[ScalarFieldRef], value: PrismaValue) -> Value<'a> {
+    convert_value(fields.first().unwrap(), value)
+}
+
 fn convert_list_value<'a>(field: &ScalarFieldRef, values: Vec<PrismaValue>) -> Value<'a> {
-    Value::Array(Some(values.into_iter().map(|val| field.value(val)).collect()))
+    Value::Array(Some(values.into_iter().map(|val| convert_value(field, val)).collect()))
```

Comment:
Offcomment: I'm surprised this doesn't break other tests.

Reply:
Why? This function does exactly the same thing? I just refactored it cause `convert_value` was already used everywhere else